go/arrow/type_traits_float16.go

```diff
@@ -0,0 +1,95 @@
+package arrow
```

Comment:
I think it would read better if the float16 implementation were in its own package (well, just the float16 implementation part, the traits-part should stay in the `arrow` package.)

WDYT?